include/miopen/miopen.h

```diff
@@ -1824,6 +1824,69 @@ MIOPEN_EXPORT miopenStatus_t miopenConvolutionBackwardBias(miopenHandle_t handle
                                                            const miopenTensorDescriptor_t dbDesc,
                                                            void* db);
 
+/* Begin of Find Mode API */
+/*! @brief Version of the Find Mode API for convolutions.
+*
+* The Find Mode API is experimental and therefore subject to change.
```

Comment:
@atamazov We agreed that this API call would not be public since we do not wish to support this API in the future and is a stop gap arrangement for MIGraphX

Reply:
I will make it. However I would appreciate if you clearly explain the reasons. What should we be afraid of?

Reply:
@JehandadKhan Done

Reply:
> we do not wish to support this API

I thought we _want_ to have means of controlling the Find Mode in our API. Of course, we do not want to support this specific public interface, because it is intermittent.

That is why I've introduced the `MIOPEN_API_VERSION_CONVOLUTION_FIND_MODE` macro. Current public interface is experimental, so we are free to change it. And if we change it, we will bump the API_VERSION macro. Or we can remove the public interface and undefine the macro. This way the application can easily find out if Find Mode API is supported in the library, and how it is supported.

Our only commitment is to keep the API_VERSION macro up to date.